src/s2wasm-main.cpp

```diff
@@ -108,7 +108,7 @@ int main(int argc, const char *argv[]) {
           : 0;
   if (options.debug) std::cerr << "Global base " << globalBase << '\n';
 
-  Linker linker(globalBase, stackAllocation, initialMem, maxMem,
+  Linker linker((Address)globalBase, (Address)stackAllocation, (Address)initialMem, (Address)maxMem,
```

Comment:
I think we should have a `toAddress` utility function that casts a `uint64_t` to and Address and checks for overflow, as it would be an error here.

We do still want `uint64_t` for those variables, since for wasm64 eventually we won't need that cast. At that point perhaps we'll have `toAddress32` and `toAddress64`?